### DISCUSSION TITLE
Add support for squash and merge in PR's

### DISCUSSION THREAD

**Discussion by ReneFroger** [magit/forge#604] [Ideas]
When a PR is approved, you can merge it in the targeted branch with `M-x forge-merge`.

But with our team we have the policy to squash all the commits and merge into the develop, for the automated changelog. So that you would see only one commit for each PR, to indicate the changes after each release.

Once you toggle `squash and merge` in Github, you see it always stays on  `squash and merge` instead of merge:

![image](https://github.com/magit/forge/assets/5211845/f8d6d88b-9413-4215-8109-2d930f07ffa2)

But it seems Forge doesn't have this option. When you merge an approved PR into the targeted branch, it gets merged without squashing. Could be another command added within forge-topic, like `M-x forge-squash-and-merge`?

**Comment by tarsius** [comment]
`forge-merge` (available at `/M` in `forge-dispatch`) supports this now.
